File: app/src/at-nhatnguyen/java/asiantech/internship/summer/javacore/Bai2.java
package asiantech.internship.summer.javacore;
import java.util.Scanner;
public class Bai2 {
    private static void doiCoSo(int n,int base){
        if(n>=base) doiCoSo(n / base, base);
        if(n % base>9) System.out.printf("%c",n%base+55);
        else
            System.out.print((n % base));
    }
    public static int input(){
        Scanner scan= new Scanner(System.in);
        boolean check= false;
        int n=0;
        while(!check){
            System.out.print(" ");
            try{
                n= scan.nextInt();
                check= true;

            }catch(Exception e){
                System.out.println("Ban phai nhap so :");
                scan.nextLine();
            }
        }
        return (n);
    }
    public static void main(String[] args) {
        System.out.println("Nhap n");
        int n= input();
        System.out.println("Nhap vao co so can chuyen sang b");
        int b= input();
        System.out.println("So " +n+ " chuyen sang co so " +b+ " thanh: ");
        doiCoSo(n,b);
    }
}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai2.kt
import java.util.*

/*
Bài 2. Viết chương trình chuyển đổi một số tự nhiên ở
hệ cơ số 10 thành số ở hệ cơ số b bất kì (1< b≤ 36).
*/
object Bai2 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter n: ")
        val n = scanner.nextInt()
        println("Enter base: ")
        val base = scanner.nextInt()
        println("Number $n from base 10 to base $base is : ")
        baseConvert(n, base)
    }

    private fun baseConvert(n: Int, base: Int) {
        if (n > base) baseConvert(n / base, base)
        if (n % base > 9) print(String.format("%c", n % base + 65 - 10))
        else print(n % base)
    }
}

